test/scripts/update-param-sys.py

```diff
@@ -0,0 +1,97 @@
+########################################################################
+# This script is designed to function with OpenC3 COSMOS's Script Runner
+########################################################################
+from openc3.script import *
+import struct
+
+# Clear counts and data products from prior tests:
+cmd("Linux_Example Ccsds_Command_Depacketizer_Instance-Reset_Counts")
+cmd("Linux_Example Command_Router_Instance-Reset_Data_Products")
+wait(1.0)
+
+# Send Noop_Arg command expecting success:
+cmd("Linux_Example Command_Router_Instance-Noop_Arg with Value 1")
+wait(1.0)
+# Check successful command count is 2 and that the last success was Noop_Arg with last Value 1:
+check_formatted("Linux_Example Software_Status_Packet Command_Router_Instance.Command_Success_Count.Value == '2'")
```

Comment:
OK. To improve this pattern for more general use I suggest the following:

- Is there a wait-on-value directive in COSMOS? Ideally you would just be waiting for the condition that the command success count is == 2, or timeout if it does not reach that? Waiting for a time is brittle and artificially slow. This could be an issue when we want to start running tests as fast as possible with randomized inputs.
- We should wrap this logic into a function that does the following: 1) Check the current value of Command_Success_Count (and Command_Fail_Count too) and save them off 2) send command 3) wait for command success count to increment and return, or time out and check command fail count to see if it has incremented 4) return value of 0 for success, 1 for fail, 2 for timeout (or something like this).

Reply:
Probably also want a function that does the opposite. If the former function was called `wait_for_cmd_success` we would also want one that is `wait_for_cmd_failure`

Reply:
> Is there a wait-on-value directive in COSMOS?
`wait_check` is similar to the `check_formatted` pattern, but adds a timeout argument. Switching to this eliminates the need for `wait` calls. Switched to `wait_check` in [9b73651](https://github.com/lasp/adamant_example/pull/33/commits/9b73651d0a7265a54769f16fb3f1d8bb6107d659).
> We should wrap this logic into a function that does the following: [...]
> Probably also want a function that does the opposite.
I will try to implement this in a future commit. `check_exception` allows for checking expressions expected to fail, although I think in all cases, if the API function call causes the script to stop then manual intervention is always required to resume or restart. This behavior may be different when not instrumenting each line.